Automattic/wp-calypso
# Externally redirect support and forums links

### Changes proposed in this Pull Request

We recently changed the support site URL from `support.wordpress.com` to `wordpress.com/support`.

Because of that change there’s currently a bug that causes `/support` links not to load when linked from within Calypso. Calypso thinks it’s an internal URL and so tries to load it as such. The main issue is that when sending users support links through Happy Chat they don’t load.

We made the same change to the forums site (forums.wordpress.com --> wordpress.com/forums)
So this would also affect /forums links.

More info: https://wp.me/pau2Xa-fv#comment-3781

---------------------

### Testing instructions

An easy way to test various links is using Happy Chat (which is also how the issue was discovered).

To do that, sign in as an operator on the Happy Chat staging HUD, here:
https://hud-staging.happychat.io/?service=https://happychat-io-staging.go-vip.co/operator
Set yourself to receive "All Chats" (you won't get live chats from users when on staging).

Then, with this branch booted up locally, go to:
http://calypso.localhost:3000/help/contact

When you start a chat as a user, you should be able to accept the chat as a Happy Chat operator, allowing you to talk to yourself.

As the operator you want to send the user links to the support and forums sites (example links below).

As the user you want to click on those links and make sure they open in a new tab with the correct URL. And NOT simply replace the URL in Calypso but not take you anywhere new.

The support and forums pages won't actually load on `calypso.localhost:3000`, but you can replace `calypso.localhost:3000` with `wordpress.com` if you want to test the pages. It shouldn't be necessary though.

**Example**

As an operator:

<img width="1537" alt="Screen Shot 2020-03-25 at 4 23 24 PM" src="https://user-images.githubusercontent.com/690843/77600025-d9348f80-6ec3-11ea-8787-a09988ad228f.png">

As a user:

<img width="862" alt="Screen Shot 2020-03-25 at 4 23 43 PM" src="https://user-images.githubusercontent.com/690843/77600099-0da84b80-6ec4-11ea-8233-a6c2fa81ba71.png">

Make sure the URLs you test use `calypso.localhost:3000`, they need to match the site the user is on.

#### URLs to check

These **should** open a new window:

http://calypso.localhost:3000/support
http://calypso.localhost:3000/support/
http://calypso.localhost:3000/support/five-step-blog-setup/
http://calypso.localhost:3000/br/support
http://calypso.localhost:3000/br/support/
http://calypso.localhost:3000/br/support/five-step-blog-setup/
http://calypso.localhost:3000/pt-br/support
http://calypso.localhost:3000/pt-br/support/
http://calypso.localhost:3000/pt-br/support/five-step-blog-setup/

http://calypso.localhost:3000/forums
http://calypso.localhost:3000/forums/
http://calypso.localhost:3000/forums/topic/website-launch-6/
http://calypso.localhost:3000/br/forums
http://calypso.localhost:3000/br/forums/
http://calypso.localhost:3000/br/forums/topic/novo-blogue/

These **should not** open a new window:

http://calypso.localhost:3000/supportt
http://calypso.localhost:3000/brr/support
http://calypso.localhost:3000/pt-brr/support
http://calypso.localhost:3000/ptt-br/support
http://calypso.localhost:3000/es/notsupport

#### Other checks

Make sure other Calypso paths work as intended (do not refresh or open a new window).
Examples:
http://calypso.localhost:3000/me
http://calypso.localhost:3000/home/[SITE]
http://calypso.localhost:3000/posts/[SITE]
http://calypso.localhost:3000/marketing/tools/[SITE]
http://calypso.localhost:3000/domains/manage/[SITE]

Click around Calypso to check for other unexpected paths not loading properly.

Make sure there are no JS errors.

#### Questions:
* Do you see any issues with the regex?
* Can you think of anything else that should be tested?
